ESP8266Interface.cpp

```diff
@@ -342,12 +344,23 @@ int ESP8266Interface::socket_close(void *handle)
 
     socket->connected = false;
     _ids[socket->id] = false;
+    _local_ports[socket->id] = 0;
```

Comment:
Is there any reason for local port not being a member of esp8266_socket?

Reply:
No need to get handle for each socket when checking if port is already taken. Btw, I could't find a better error code than NSAPI_ERROR_PARAMETER for the case that port is already reserved